airflow/models/dag.py

```diff
@@ -458,6 +458,101 @@ def previous_schedule(self, dttm):
         elif self.normalized_schedule_interval is not None:
             return timezone.convert_to_utc(dttm - self.normalized_schedule_interval)
 
+    def next_dagrun_info(self, date_last_automated_dagrun : Optional[pendulum.DateTime]):
+        """
+        Get information about the next DagRun of this dag after ``date_last_automated_dagrun`` -- the
+        execution date, and the earliest it could be scheduled
+
+        :param date_last_automated_dagrun: The max(execution_date) of existing
+            "automated" DagRuns for this dag (scheduled or backfill, but not
+            manual)
+        """
+        next_execution_date = self.next_dagrun_after_date(date_last_automated_dagrun)
+
+        if next_execution_date is None or self.schedule_interval in (None, '@once'):
+            return None
+
+        return {
+            'execution_date': next_execution_date,
+            'can_be_created_after': self.following_schedule(next_execution_date)
+        }
```

Comment:
Not sure, but may be cleaner to return a tuple instead. Say line 473 will be `return None, None`, and line 475 becomes `return next_execution_date, self.following_schedule(next_execution_date)`

In such a way, calling also becomes simpler

```python
orm_dag.next_dagrun, orm_dag.next_dagrun_create_after = dag.next_dagrun_info(xxx)
```

Of course quite subjective opinion, so feel free to ignore

Reply:
Nice idea, will do.